macumba.ServerError: (ServerError(...) - cannot assign unit X to machine N: series does not match

With Vivid-Kilo chosen, the openstack-installer deployment indicates a series mismatch, which results in an infinite deploy process.

macumba.ServerError: (ServerError(...), 'cannot assign unit "neutron-gateway/0" to machine 2: series does not match')

```
[Services]
NAME                      STATUS  EXPOSED CHARM
glance                    blocked false   cs:trusty/glance-28
glance-simplestreams-sync unknown false   cs:trusty/glance-simplestreams-sync-7
keystone                  blocked false   cs:trusty/keystone-31
mysql                     unknown false   cs:trusty/mysql-30
neutron-api               blocked false   cs:trusty/neutron-api-21
neutron-gateway           unknown false   cs:trusty/neutron-gateway-7
neutron-openvswitch               false   cs:trusty/neutron-openvswitch-13
nova-cloud-controller     blocked false   cs:trusty/nova-cloud-controller-64
nova-compute              unknown false   cs:trusty/nova-compute-33
ntp                               false   cs:trusty/ntp-14
openstack-dashboard       blocked false   cs:trusty/openstack-dashboard-19
rabbitmq-server           active  false   cs:trusty/rabbitmq-server-42

[Units]
ID                          WORKLOAD-STATE AGENT-STATE VERSION MACHINE PORTS                               PUBLIC-ADDRESS MESSAGE
glance-simplestreams-sync/0 unknown        idle        1.25.0  3/lxc/5                                     10.245.173.46
glance/0                    blocked        idle        1.25.0  3/lxc/6 9292/tcp                            10.245.173.38  Missing relations: identity, database
keystone/0                  blocked        idle        1.25.0  3/lxc/1                                     10.245.173.112 Missing relations: database                                                                                                                      
mysql/0                     unknown        idle        1.25.0  3/lxc/0                                     10.245.173.110
neutron-api/0               blocked        idle        1.25.0  3/lxc/7 9696/tcp                            10.245.173.42  Missing relations: messaging, identity, database
neutron-gateway/0           unknown        allocating                                                                     Waiting for agent initialization to finish
nova-cloud-controller/0     blocked        idle        1.25.0  3/lxc/3 3333/tcp,8773/tcp,8774/tcp,9696/tcp 10.245.173.200 Missing relations: messaging, image, compute, identity, database
nova-compute/0              unknown        allocating                                                                     Waiting for agent initialization to finish
openstack-dashboard/0       blocked        idle        1.25.0  3/lxc/4 80/tcp,443/tcp                      10.245.173.45  Missing relations: identity
rabbitmq-server/0           active         idle        1.25.0  3/lxc/2 5672/tcp                            10.245.171.218 Unit is ready

[Machines]
ID         STATE   VERSION DNS                         INS-ID                                                         SERIES HARDWARE
0          started 1.25.0  fat-machine.dellstack       /MAAS/api/1.0/nodes/node-d4692494-8228-11e4-8078-d4bed9a84493/ vivid  arch=amd64 cpu-cores=8 mem=49152M tags=node-d4692494-8228-11e4-8078-d4bed9a84493,uosci,storage
1          started 1.25.0  cylindrical-base.dellstack  /MAAS/api/1.0/nodes/node-d65842d0-8228-11e4-9105-d4bed9a84493/ vivid  arch=amd64 cpu-cores=8 mem=49152M tags=node-d65842d0-8228-11e4-9105-d4bed9a84493,uosci,gateway
2          started 1.25.0  grizzled-family.dellstack   /MAAS/api/1.0/nodes/node-d74ee982-8228-11e4-8078-d4bed9a84493/ vivid  arch=amd64 cpu-cores=8 mem=49152M tags=node-d74ee982-8228-11e4-8078-d4bed9a84493,uosci,storage,object_zone_1,object_storage
3          started 1.25.0  imaginative-error.dellstack /MAAS/api/1.0/nodes/node-d7cb18b8-8228-11e4-9105-d4bed9a84493/ vivid  arch=amd64 cpu-cores=8 mem=49152M tags=node-d7cb18b8-8228-11e4-9105-d4bed9a84493,uosci,storage,object_zone_2,object_storage
```




```
jenkins@juju-osci-machine-13:~⟫ juju debug-log
machine-0: 2015-11-23 15:53:34 ERROR juju.state.unit unit.go:743 unit nova-compute/0 cannot get assigned machine: unit "nova-compute/0" is not assigned to a machine
machine-0: 2015-11-23 15:53:34 WARNING juju.apiserver.client status.go:677 error fetching public address: unit nova-compute/0 cannot get assigned machine: unit "nova-compute/0" is not assigned to a machine
machine-0: 2015-11-23 15:53:35 ERROR juju.state.unit unit.go:743 unit neutron-gateway/0 cannot get assigned machine: unit "neutron-gateway/0" is not assigned to a machine
machine-0: 2015-11-23 15:53:35 WARNING juju.apiserver.client status.go:677 error fetching public address: unit neutron-gateway/0 cannot get assigned machine: unit "neutron-gateway/0" is not assigned to a machine
machine-0: 2015-11-23 15:53:35 ERROR juju.state.unit unit.go:743 unit nova-compute/0 cannot get assigned machine: unit "nova-compute/0" is not assigned to a machine
machine-0: 2015-11-23 15:53:35 WARNING juju.apiserver.client status.go:677 error fetching public address: unit nova-compute/0 cannot get assigned machine: unit "nova-compute/0" is not assigned to a machine
machine-0: 2015-11-23 15:53:44 ERROR juju.state.unit unit.go:743 unit nova-compute/0 cannot get assigned machine: unit "nova-compute/0" is not assigned to a machine
machine-0: 2015-11-23 15:53:44 WARNING juju.apiserver.client status.go:677 error fetching public address: unit nova-compute/0 cannot get assigned machine: unit "nova-compute/0" is not assigned to a machine
machine-0: 2015-11-23 15:53:44 ERROR juju.state.unit unit.go:743 unit neutron-gateway/0 cannot get assigned machine: unit "neutron-gateway/0" is not assigned to a machine
machine-0: 2015-11-23 15:53:44 WARNING juju.apiserver.client status.go:677 error fetching public address: unit neutron-gateway/0 cannot get assigned machine: unit "neutron-gateway/0" is not assigned to a machine
machine-0: 2015-11-23 15:53:55 ERROR juju.state.unit unit.go:743 unit neutron-gateway/0 cannot get assigned machine: unit "neutron-gateway/0" is not assigned to a machine
machine-0: 2015-11-23 15:53:55 WARNING juju.apiserver.client status.go:677 error fetching public address: unit neutron-gateway/0 cannot get assigned machine: unit "neutron-gateway/0" is not assigned to a machine
machine-0: 2015-11-23 15:53:55 ERROR juju.state.unit unit.go:743 unit nova-compute/0 cannot get assigned machine: unit "nova-compute/0" is not assigned to a machine
machine-0: 2015-11-23 15:53:55 WARNING juju.apiserver.client status.go:677 error fetching public address: unit nova-compute/0 cannot get assigned machine: unit "nova-compute/0" is not assigned to a machine
machine-0: 2015-11-23 15:53:56 ERROR juju.state.unit unit.go:743 unit neutron-gateway/0 cannot get assigned machine: unit "neutron-gateway/0" is not assigned to a machine
machine-0: 2015-11-23 15:53:56 WARNING juju.apiserver.client status.go:677 error fetching public address: unit neutron-gateway/0 cannot get assigned machine: unit "neutron-gateway/0" is not assigned to a machine
machine-0: 2015-11-23 15:53:56 ERROR juju.state.unit unit.go:743 unit nova-compute/0 cannot get assigned machine: unit "nova-compute/0" is not assigned to a machine
machine-0: 2015-11-23 15:53:56 WARNING juju.apiserver.client status.go:677 error fetching public address: unit nova-compute/0 cannot get assigned machine: unit "nova-compute/0" is not assigned to a machine
```




```
[INFO: 11-23 06:44:22] Starting OpenStack Installer v0.99.26
[INFO: 11-23 06:44:22] Start command: ['/usr/bin/openstack-install', '--debug', '--headless', '--openstack-release', 'kilo', '--series', 'vivid', '--constraints', 'arch=amd64 tags=uosci', '-c', '/var/lib/jenkins/osi-config-multi.yaml']
[INFO: 11-23 06:44:22] Creating juju directories: /var/lib/jenkins/.cloud-install/juju
[INFO: 11-23 06:44:24] Running Kilo release
[INFO: 11-23 06:44:24] Performing a Multi install with existing MAAS
[INFO: 11-23 06:44:24] pollinate: sudo su - -c 'pollinate -q -r --curl-opts "--user-agent uoi/e590e777-ec1f-4148-a2a0-7e63cbf2c1fc/IM"'
stdin: is not a tty
[INFO: 11-23 06:44:25] Bootstrapping Juju
[INFO: 11-23 06:44:25] Bootstrapping Juju
[INFO: 11-23 06:55:25] Starting deployment of OpenStack
[INFO: 11-23 06:55:40] Waiting for 3 maas machines to be ready. Machines Summary: 1 deployed, 6 ready
[INFO: 11-23 06:55:45] Waiting for machines to start: 3 unknown
[INFO: 11-23 06:55:51] Waiting for machines to start: 1 pending, 2 unknown
[INFO: 11-23 06:56:03] Waiting for machines to start: 2 pending, 1 unknown
[INFO: 11-23 06:56:09] Waiting for machines to start: 3 pending
[INFO: 11-23 07:04:05] Waiting for machines to start: 2 pending, 1 down (started)
[INFO: 11-23 07:04:26] Waiting for machines to start: 1 pending, 2 down (started)
[INFO: 11-23 07:04:29] Waiting for machines to start: 1 pending, 2 started
[INFO: 11-23 07:04:33] Waiting for machines to start: 1 down (started), 2 started
[INFO: 11-23 07:04:59] Verifying service deployments
[INFO: 11-23 07:04:59] Pending charms to deploy: MySQL, NTP, Keystone, RabbitMQ Server, Controller, Neutron, Openstack Dashboard, Neutron OpenVSwitch, Glance - Simplestreams Image Sync, Glance, Compute, Neutron API
[INFO: 11-23 07:04:59] Checking if MySQL is deployed
[INFO: 11-23 07:04:59] Deploying MySQL to machine lxc:3
[INFO: 11-23 07:05:08] Deployed MySQL.
[INFO: 11-23 07:05:08] Pending charms to deploy: NTP, Keystone, RabbitMQ Server, Controller, Neutron, Openstack Dashboard, Neutron OpenVSwitch, Glance - Simplestreams Image Sync, Glance, Compute, Neutron API
[INFO: 11-23 07:05:08] Checking if NTP is deployed
[INFO: 11-23 07:05:08] Deploying NTP
[INFO: 11-23 07:05:12] Deployed NTP.
[INFO: 11-23 07:05:15] Pending charms to deploy: Keystone, RabbitMQ Server, Controller, Neutron, Openstack Dashboard, Neutron OpenVSwitch, Glance - Simplestreams Image Sync, Glance, Compute, Neutron API
[INFO: 11-23 07:05:15] Checking if Keystone is deployed
[INFO: 11-23 07:05:16] Deploying Keystone to machine lxc:3
[INFO: 11-23 07:05:24] Deployed Keystone.
[INFO: 11-23 07:05:24] Pending charms to deploy: RabbitMQ Server, Controller, Neutron, Openstack Dashboard, Neutron OpenVSwitch, Glance - Simplestreams Image Sync, Glance, Compute, Neutron API
[INFO: 11-23 07:05:24] Checking if RabbitMQ Server is deployed
[INFO: 11-23 07:05:24] Deploying RabbitMQ Server to machine lxc:3
[INFO: 11-23 07:05:32] Deployed RabbitMQ Server.
[INFO: 11-23 07:05:32] Pending charms to deploy: Controller, Neutron, Openstack Dashboard, Neutron OpenVSwitch, Glance - Simplestreams Image Sync, Glance, Compute, Neutron API
[INFO: 11-23 07:05:32] Checking if Controller is deployed
[INFO: 11-23 07:05:32] Deploying Controller to machine lxc:3
[INFO: 11-23 07:05:41] Deployed Controller.
[INFO: 11-23 07:05:43] Pending charms to deploy: Neutron, Openstack Dashboard, Neutron OpenVSwitch, Glance - Simplestreams Image Sync, Glance, Compute, Neutron API
[INFO: 11-23 07:05:43] Checking if Neutron is deployed
[INFO: 11-23 07:05:43] Deploying Neutron to machine 2
[ERROR: 11-23 07:05:50] Error deploying
Traceback (most recent call last):
  File "/usr/share/openstack/cloudinstall/charms/__init__.py", line 272, in deploy
    config_yaml, self.constraints, machine_spec)
  File "/usr/share/openstack/macumba/__init__.py", line 422, in deploy
    Params=dict(params)))
  File "/usr/share/openstack/macumba/__init__.py", line 284, in call
    return self.receive(req_id, timeout)
  File "/usr/share/openstack/macumba/__init__.py", line 268, in receive
    raise ServerError(res['Error'], res)
macumba.ServerError: (ServerError(...), 'cannot assign unit "neutron-gateway/0" to machine 2: series does not match')
[INFO: 11-23 07:05:55] Pending charms to deploy: Neutron, Openstack Dashboard, Neutron OpenVSwitch, Glance - Simplestreams Image Sync, Glance, Compute, Neutron API
[INFO: 11-23 07:05:55] Pending charms to deploy: Neutron, Openstack Dashboard, Neutron OpenVSwitch, Glance - Simplestreams Image Sync, Glance, Compute, Neutron API
[INFO: 11-23 07:05:55] Checking if Neutron is deployed
[INFO: 11-23 07:05:55] Deploying Neutron to machine 2
[ERROR: 11-23 07:05:56] Error deploying
Traceback (most recent call last):
  File "/usr/share/openstack/cloudinstall/charms/__init__.py", line 272, in deploy
    config_yaml, self.constraints, machine_spec)
  File "/usr/share/openstack/macumba/__init__.py", line 422, in deploy
    Params=dict(params)))
  File "/usr/share/openstack/macumba/__init__.py", line 284, in call
    return self.receive(req_id, timeout)
  File "/usr/share/openstack/macumba/__init__.py", line 268, in receive
    raise ServerError(res['Error'], res)
macumba.ServerError: (ServerError(...), 'cannot add service "neutron-gateway": service already exists')
[INFO: 11-23 07:06:01] Pending charms to deploy: Neutron, Openstack Dashboard, Neutron OpenVSwitch, Glance - Simplestreams Image Sync, Glance, Compute, Neutron API
[INFO: 11-23 07:06:01] Pending charms to deploy: Neutron, Openstack Dashboard, Neutron OpenVSwitch, Glance - Simplestreams Image Sync, Glance, Compute, Neutron API
[INFO: 11-23 07:06:01] Checking if Neutron is deployed
[INFO: 11-23 07:06:01] Deploying Neutron to machine 2
[ERROR: 11-23 07:06:01] Error deploying
Traceback (most recent call last):
  File "/usr/share/openstack/cloudinstall/charms/__init__.py", line 272, in deploy
    config_yaml, self.constraints, machine_spec)
  File "/usr/share/openstack/macumba/__init__.py", line 422, in deploy
    Params=dict(params)))
  File "/usr/share/openstack/macumba/__init__.py", line 284, in call
    return self.receive(req_id, timeout)
  File "/usr/share/openstack/macumba/__init__.py", line 268, in receive
    raise ServerError(res['Error'], res)
macumba.ServerError: (ServerError(...), 'cannot add service "neutron-gateway": service already exists')
[INFO: 11-23 07:06:06] Pending charms to deploy: Neutron, Openstack Dashboard, Neutron OpenVSwitch, Glance - Simplestreams Image Sync, Glance, Compute, Neutron API
[INFO: 11-23 07:06:06] Pending charms to deploy: Neutron, Openstack Dashboard, Neutron OpenVSwitch, Glance - Simplestreams Image Sync, Glance, Compute, Neutron API
[INFO: 11-23 07:06:06] Checking if Neutron is deployed
[INFO: 11-23 07:06:07] Neutron is already deployed, skipping
[INFO: 11-23 07:06:07] Checking if Openstack Dashboard is deployed
[INFO: 11-23 07:06:10] Deploying Openstack Dashboard to machine lxc:3
[INFO: 11-23 07:06:19] Deployed Openstack Dashboard.
[INFO: 11-23 07:06:19] Pending charms to deploy: Neutron OpenVSwitch, Glance - Simplestreams Image Sync, Glance, Compute, Neutron API
[INFO: 11-23 07:06:19] Checking if Neutron OpenVSwitch is deployed
[INFO: 11-23 07:06:19] Deploying Neutron OpenVSwitch
[INFO: 11-23 07:06:24] Deployed Neutron OpenVSwitch.
[INFO: 11-23 07:06:24] Pending charms to deploy: Glance - Simplestreams Image Sync, Glance, Compute, Neutron API
[INFO: 11-23 07:06:24] Checking if Glance - Simplestreams Image Sync is deployed
[INFO: 11-23 07:06:24] Deploying Glance - Simplestreams Image Sync to machine lxc:3
[INFO: 11-23 07:06:33] Deployed Glance - Simplestreams Image Sync.
[INFO: 11-23 07:06:35] Pending charms to deploy: Glance, Compute, Neutron API
[INFO: 11-23 07:06:35] Checking if Glance is deployed
[INFO: 11-23 07:06:35] Deploying Glance to machine lxc:3
[INFO: 11-23 07:06:46] Deployed Glance.
[INFO: 11-23 07:06:46] Pending charms to deploy: Compute, Neutron API
[INFO: 11-23 07:06:46] Checking if Compute is deployed
[INFO: 11-23 07:06:46] Deploying Compute to machine 1
[ERROR: 11-23 07:06:55] Error deploying
Traceback (most recent call last):
  File "/usr/share/openstack/cloudinstall/charms/__init__.py", line 272, in deploy
    config_yaml, self.constraints, machine_spec)
  File "/usr/share/openstack/macumba/__init__.py", line 422, in deploy
    Params=dict(params)))
  File "/usr/share/openstack/macumba/__init__.py", line 284, in call
    return self.receive(req_id, timeout)
  File "/usr/share/openstack/macumba/__init__.py", line 268, in receive
    raise ServerError(res['Error'], res)
macumba.ServerError: (ServerError(...), 'cannot assign unit "nova-compute/0" to machine 1: series does not match')
[INFO: 11-23 07:07:00] Pending charms to deploy: Compute, Neutron API
[INFO: 11-23 07:07:00] Pending charms to deploy: Compute, Neutron API
[INFO: 11-23 07:07:00] Checking if Compute is deployed
[INFO: 11-23 07:07:02] Deploying Compute to machine 1
[ERROR: 11-23 07:07:03] Error deploying
Traceback (most recent call last):
  File "/usr/share/openstack/cloudinstall/charms/__init__.py", line 272, in deploy
    config_yaml, self.constraints, machine_spec)
  File "/usr/share/openstack/macumba/__init__.py", line 422, in deploy
    Params=dict(params)))
  File "/usr/share/openstack/macumba/__init__.py", line 284, in call
    return self.receive(req_id, timeout)
  File "/usr/share/openstack/macumba/__init__.py", line 268, in receive
    raise ServerError(res['Error'], res)
macumba.ServerError: (ServerError(...), 'cannot add service "nova-compute": service already exists')
[INFO: 11-23 07:07:08] Pending charms to deploy: Compute, Neutron API
[INFO: 11-23 07:07:08] Pending charms to deploy: Compute, Neutron API
[INFO: 11-23 07:07:08] Checking if Compute is deployed
[INFO: 11-23 07:07:08] Compute is already deployed, skipping
[INFO: 11-23 07:07:08] Checking if Neutron API is deployed
[INFO: 11-23 07:07:08] Deploying Neutron API to machine lxc:3
[INFO: 11-23 07:07:14] Deployed Neutron API.
[INFO: 11-23 07:07:14] Waiting for deployed services to be in a ready state.
[INFO: 11-23 07:07:15] Checking availability of glance-simplestreams-sync:pending, glance:pending, openstack-dashboard:pending, neutron-gateway:pending, neutron-api:pending, nova-cloud-controller:pending, mysql:pending, rabbitmq-server:pending, keystone:pending, nova-compute:pending 
[INFO: 11-23 07:12:35] Checking availability of glance-simplestreams-sync:pending, glance:pending, openstack-dashboard:pending, neutron-gateway:pending, neutron-api:pending, nova-cloud-controller:pending, rabbitmq-server:pending, keystone:pending, nova-compute:pending 
[INFO: 11-23 07:13:39] Checking availability of glance-simplestreams-sync:pending, glance:pending, openstack-dashboard:pending, neutron-gateway:pending, neutron-api:pending, nova-cloud-controller:pending, keystone:pending, nova-compute:pending 
[INFO: 11-23 07:14:00] Checking availability of glance-simplestreams-sync:pending, glance:pending, openstack-dashboard:pending, neutron-gateway:pending, neutron-api:pending, nova-compute:pending 
[INFO: 11-23 07:14:21] Checking availability of glance-simplestreams-sync:pending, openstack-dashboard:pending, neutron-gateway:pending, nova-compute:pending 
[INFO: 11-23 07:14:43] Checking availability of glance-simplestreams-sync:pending, neutron-gateway:pending, nova-compute:pending 
[INFO: 11-23 07:16:29] Checking availability of neutron-gateway:pending, nova-compute:pending 
```
And it is eternally looping at this point, 9 hrs, not exiting with a timeout.

--- a/cloudinstall/charms/__init__.py
+++ b/cloudinstall/charms/__init__.py
@@ -270,9 +270,10 @@ export OS_REGION_NAME=\"RegionOne\"
 
             self.juju.deploy(_charm_name_rev, self.charm_name, num_units,
                              config_yaml, self.constraints, machine_spec)
-        except MacumbaError:
-            log.exception("Error deploying")
-            return True
+        except MacumbaError as e:
+            log.exception("Error deploying: {}".format(e))
+            raise Exception(
+                "Error deploying: {} {}".format(self.charm_name, e))
 
         self.ui.status_info_message("Deployed {0}.".format(self.display_name))
         return False

